src/main/java/com/louis/HashSet.java

```diff
@@ -0,0 +1,186 @@
+        while (set[index] != null && !element.equals(set[index].getElement())) {
+            index = (index + 1) % currentCapacity;
+        }
+
+        // Increases the size if a new element is being put in.
+        if (set[index] == null) {
+            currentSize++;
+        }
+        set[index] = new HashEntry(element);
+
+        resizeIfRequired();
+    }
+
+    /**
+     * Hashs an element.
+     *
+     * The location of the element in the set is equivalent to this.
+     * 
+     * @param element is the element to be hashed.
+     * @return Returns the hash of the element.
+     */
+    private int hash(E element) {
+        int hashValue = element.hashCode();
+        hashValue ^= (hashValue >>> 20) ^ (hashValue >>> 12);
+        hashValue = hashValue ^ (hashValue >>> 7) ^ (hashValue >>> 4);
+        return Math.abs(hashValue);
+    }
+
+    /**
+     * Resizes the underlying array if required.
+     */
+    private void resizeIfRequired() {
+        if (!((currentSize < currentCapacity * MIN_LOAD && currentCapacity > MIN_CAPACITY)
+                || currentSize > currentCapacity * MAX_LOAD)) {
+            return;
+        }
+        int newCapacity = (int) (currentSize / RESIZE_LOAD);
+
+        @SuppressWarnings("unchecked")
+        // Makes the new array.
+        HashEntry[] newArray = (HashEntry[]) Array.newInstance(HashEntry.class, newCapacity);
+        // HashEntry[] newArray = new HashEntry[newCapacity];
+
+        for (int oldIndex = 0; oldIndex < currentCapacity; oldIndex++) {
+            HashEntry oldEntry = set[oldIndex];
+            if (oldEntry == null) {
+                continue;
+            }
+            int index = hash(oldEntry.getElement()) % newCapacity;
+            while (newArray[index] != null
+                    && !oldEntry.getElement().equals(newArray[index].getElement())) {
+                // Gets the next index.
+                index = (index + 1) % newCapacity;
+            }
+            newArray[index] = oldEntry;
+        }
+        set = newArray;
+        currentCapacity = newCapacity;
+    }
+
+    /**
+     * Removes the element from the hashtable.
+     * 
+     * @param element is the element to be deleted.
+     * @throws IllegalArgumentException is thrown if element is null.
+     */
+    public void remove(E element) {
+        Preconditions.checkArgument(element != null, "First argument to remove(E element) is null");
+        List<HashEntry> entries = new ArrayList<HashEntry>();
+
+        // Locates the element.
+        int index = hash(element) % currentCapacity;
+        while (set[index] != null && !element.equals(set[index].getElement())) {
+            index = (index + 1) % currentCapacity;
+            if (set[index] == null)
+                System.out.printf("Element %s already deleted %n", element.toString());
+        }
+
+        // Extracts elements that collided with this element.
+        while (set[index] != null) {
```

Comment:
Refactor this block to a separate method called `extractContiguousElements` or something similar.